File: javascript/easy/bestTimeToBuyAndSellStock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProfit = 0, i = 0, j = 1;
    while(j < prices.length) {
        maxProfit = Math.max(maxProfit, (prices[j] - prices[i]));
        if(prices[j] < prices[i]) {
            i = j;
        }
        j++;
    }
    return maxProfit;
};

File: ruby/easy/best_time_to_buy_and_sell_stock.rb
# @param {Integer[]} prices
# @return {Integer}
def max_profit(prices)
    i, max_profit = 0, 0
    prices.each_with_index do |price, j|
        max_profit = [max_profit, (price - prices[i])].max
        i = j if(price < prices[i])
    end
    return max_profit
end
